plugins/module_utils/pulp.py

```diff
@@ -161,7 +161,9 @@ def find(self, failsafe=True, parameters=None):
         parameters["limit"] = 1
         parameters.update(self.natural_key)
         search_result = self.module.pulp_api.call(self._list_id, parameters=parameters)
-        if search_result["count"] == 1:
+        # While we limit to one result, the count may be more than one.
+        # This may happen if the parameters are not unique, e.g. for a publication.
+        if search_result["count"] >= 1:
```

Comment:
I think we should fail if we find more than one item.

Reply:
I wondered if you might say this.

Reply:
At least we agree that creating the third publication is the worst behaviour.

Reply:
I was 50/50 on which way to go with the solution tbh